### DISCUSSION TITLE
WYSIWYG editor is deleting the code

### DISCUSSION THREAD

**Discussion by silbox-ch** [GetPublii/Publii#1572] [Help]
Hello, I'm trying to add this code to a button on my page.
So I have <button></button>
then I edit source code, and add
onclick="$crisp.push(['do', 'chat:open'])"
now it's 

`<button onclick="$crisp.push(['do', 'chat:open'])">Click here</button>`

When I apply the modifications, the editor delete what I just added.

and I don't know if it helps, but how do I change from WYSIWYG to others editors for an existing article?

**Comment by candidexmedia** [comment]
> When I apply the modifications, the editor delete what I just added.

See: https://github.com/GetPublii/Publii/discussions/1516 - Usage of custom elements in the WYSIWYG editor 

> and I don't know if it helps, but how do I change from WYSIWYG to others editors for an existing article?

It's not possible to do, unfortunately. You'll have to copy paste.

**Comment by silbox-ch** [comment]
Thank you @candidexmedia 
Solution was there https://getpublii.com/docs/website-settings.html#editors

In my case, needed to add button[*]

![image](https://github.com/user-attachments/assets/361aecbc-fbcc-469e-a550-3f90b3f532f3)
